webservice/components/webservice_component.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2020 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import requests
+
+from odoo.addons.component.core import AbstractComponent, Component
+
+
+class BaseWebServiceAdapter(AbstractComponent):
+    _name = "base.webservice.adapter"
+    _collection = "webservice.backend"
+
+
+class BaseRestRequestsAdapter(Component):
+    _name = "base.requests"
+    _usage = "webservice.httprequest"
+    _inherit = "base.webservice.adapter"
+
+    def get(self, params=None, timeout=20, verify=None, cookies=None, **kwargs):
+        request = requests.get(
+            self.collection.url.format(**kwargs),
+            params=params,
+            auth=self._get_auth(**kwargs),
+            cookies=cookies,
+            verify=verify,
+            headers=self._get_headers(**kwargs),
+            timeout=timeout,
+        )
+        request.raise_for_status()
+        return request.content
+
+    def post(self, data, timeout=20, verify=None, cookies=None, **kwargs):
+        request = requests.post(
```

Comment:
just use `request.request` for all kind of requests and move the logic to `_make_request` method. See https://requests.readthedocs.io/en/master/api/#main-interface

Reply:
I made some changes. I hope it fills your expectations :wink: